src/main/java/org/example/Person.java

```diff
@@ -0,0 +1,106 @@
+package org.example;
+
+abstract public class Person {
+    private String firstName;
+    private String lastName;
+    private int age;
+    private Person spouse = null;
+
+    public Person(String firstName, String lastName, int age) throws Exception {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        validateAge(age);
+        this.age = age;
+    }
+
+    public Person(String firstName, String lastName, int age, Person spouse) throws Exception {
+         this(firstName, lastName, age);
```

Comment:
Better if `Person(String,String, int)` will be calling constructor `Person(String,String, int, Person)` but not vice versa because logic of setting and processing variables should be in the same place you can just call with `null` argument at the end